lib/Dialect/FIRRTL/Transforms/PrefixModules.cpp

```diff
@@ -0,0 +1,177 @@
+//===- PrefixModules.cpp - Prefix module names pass -------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the PrefixModules pass.
+//
+//===----------------------------------------------------------------------===//
+
+#include "./PassDetails.h"
+#include "circt/Dialect/FIRRTL/FIRRTLAnnotations.h"
+#include "circt/Dialect/FIRRTL/FIRRTLOps.h"
+#include "circt/Dialect/FIRRTL/InstanceGraph.h"
+#include "circt/Dialect/FIRRTL/Passes.h"
+#include "circt/Support/LLVM.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/PostOrderIterator.h"
+#include "llvm/ADT/SmallSet.h"
+
+using namespace circt;
+using namespace firrtl;
+
+namespace {
+
+/// This is the prefix which will be applied to a module.
+struct PrefixInfo {
+
+  /// The string to prefix on to the module and all of its children.
+  StringRef prefix;
+
+  /// If true, this prefix applies to the module itself.  If false, the prefix
+  /// only applies to it's children.
+  bool inclusive;
+};
+
+/// This maps a FModuleOp to a list of all prefixes that need to be applied.
+/// When a module has multiple prefixes, it will be cloned for each one.
+using PrefixMap = DenseMap<StringRef, llvm::SmallSet<std::string, 1>>;
+
+/// Get the PrefixInfo for a module. This reads the annotations on module.  If
+/// the module is not annotated, the prefix returned will be empty.
+PrefixInfo getPrefixInfo(FModuleOp module) {
```

Comment:
Plz mark function static to shrink the scope of the anonymous namespace.  It gives better "local reasoning" when reading source code.